infamous_imps/server/dba.py

```diff
@@ -0,0 +1,94 @@
+from os.path import exists
+import sqlite3
+
+def start():
+        """ This programmm will check if database exists in server if not create it"""
+        if(exists("data.db")):
+            pass
+        else:
+            f = open("data.db","w")
+            f.close()
+            con = sqlite3.connect("data.db")
+            cur = con.cursor()
+            cur.execute('''CREATE TABLE chat
+               (user text,timestamp real,message text)''')
+            cur.execute('''CREATE TABLE auth
+               (user text, password text)''')
+            con.commit()
+            con.close()
+start()
+
+
+def add_message(server,timestamp,user,text):
+    """ Adds new message to the server"""
+    con = sqlite3.connect("data.db") 
+    cur = con.cursor()
+    cur.execute("insert into chat values (?,?,?)", (user,timestamp,text))
+    con.commit()
+    con.close()
+        
+        
+def add_user(name,hashed):
+    "Adds new player to the database"""
+    con = sqlite3.connect("data.db") 
+    cur = con.cursor()
+    cur.execute("insert into auth values (?,?)", (name,hashed))
+    con.commit()
+    con.close()
+        
+        
+def delete_user(name,hashed):
+    """Deletes the user from the database"""
+    con = sqlite3.connect("data.db") 
+    cur = con.cursor()
+    print(name)
+    name = (name,)
+    cur.execute('DELETE from auth where user =?',name)
+    con.commit()
+    con.close()
+        
+def see_message(server):
+    """See all message from the server"""
+    con = sqlite3.connect("data.db") 
+    cur = con.cursor()
+    return cur.execute('SELECT * FROM server ORDER BY timestamp')
+def check_user(name,hashed):
+    """Check if given credentials are correct"""
+    con = sqlite3.connect("data.db") 
+    cur = con.cursor()
+    name = (name,)
+    c = cur.execute(f'SELECT * FROM auth where user = ?',name)

```

Comment:
This is the wrong syntax for f-string
Example: `print(f"{val}for{val} is a portal for {val}.")`

Rather remove 'f' from it if you do not want interpolation.